File: delightreading-server/src/main/java/com/delightreading/user/UserAuthenticationRepository.java
package com.delightreading.user;

import com.delightreading.user.model.UserAuthenticationEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserAuthenticationRepository extends JpaRepository<UserAuthenticationEntity, Long> {

    Optional<UserAuthenticationEntity> findByUid(String uid);

    Optional<UserAuthenticationEntity> findByProviderAndProviderAccountId(String provider, String providerAccountId);
}

File: delightreading-backend/src/main/kotlin/com/delightreading/user/UserAuthenticationRepository.kt
package com.delightreading.user

import com.delightreading.user.model.UserAuthenticationEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserAuthenticationRepository : JpaRepository<UserAuthenticationEntity, Long> {

    fun findByUid(uid: String): UserAuthenticationEntity?

    fun findByProviderAndProviderAccountId(
        provider: String,
        providerAccountId: String
    ): UserAuthenticationEntity?

}